crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs

```diff
@@ -166,10 +167,20 @@ impl OffChainDatabase for OffChainIterableKeyValueView {
             .into_owned())
     }
 
-    fn old_transaction(&self, id: &TxId) -> StorageResult<Option<Transaction>> {
+    fn old_transaction(&self, id: &TxId) -> StorageResult<Transaction> {
         self.storage_as_ref::<OldTransactions>()
-            .get(id)
-            .map(|tx| tx.map(|tx| tx.into_owned()))
+            .get(id)?
+            .ok_or(not_found!(OldTransactions))
+            .map(|tx| tx.into_owned())
+    }
+
+    fn old_transactions<'a>(
+        &'a self,
+        ids: BoxedIter<'a, &'a TxId>,
+    ) -> BoxedIter<'a, StorageResult<Transaction>> {
+        <Self as StorageBatchInspect<OldTransactions>>::get_batch(self, ids)
+            .map(|result| result.and_then(|opt| opt.ok_or(not_found!(OldTransactions))))
+            .into_boxed()
```

Comment:
If you want, you can add the same syntax sugars that we did with `storage_as_ref` to avoid `<Self as StorageInspect<M>>` usage.

Reply:
Oh interesting, that would be nice. I'll look into it.

Reply:
Turns out this is not as trivial as I had thought, as we're hitting some lifetime issues. 

### The problem
`get_batch` returns an iterator bound to the lifetime of the `self` parameter. I.e. the storage we call it on. This is necessary since we do `self.get(...)` within the `KeyValueInspect::get_batch` implementation.

So while we can implement `StorageBatchInspect` for the `StorageRef` type as:

```rust
impl<'a, S, Type> StorageBatchInspect<Type> for StorageRef<'a, S, Type>
where
    S: StorageBatchInspect<Type>,
    Type: Mappable,
{
    #[inline(always)]
    fn get_batch<'b, Iter>(
        &'b self,
        _keys: Iter,
    ) -> impl Iterator<Item = Result<Option<Type::OwnedValue>>> + 'b
    where
        Iter: 'b + IntoIterator<Item = &'b Type::Key>,
        Type::Key: 'b,
    {
        None.into_iter() // Note that we'd need access to `self.0` here which requires VM changes, or copying the `StorageRef` type into this crate. But that's a separate, and very manageable issue
    }
}
```

when we try to use it as

```rust
    fn old_transactions<'a>(
        &'a self,
        ids: BoxedIter<'a, &'a TxId>,
    ) -> BoxedIter<'a, StorageResult<Transaction>> {
        self.storage::<OldTransactions>()
            .get_batch(ids)
            .map(|result| result.and_then(|opt| opt.ok_or(not_found!(OldTransactions))))
            .into_boxed()
    }
```

we're hitting this issue
```
error[E0716]: temporary value dropped while borrowed
   --> crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs:183:9
    |
179 |       fn old_transactions<'a>(
    |                           -- lifetime `'a` defined here
...
183 |           self.storage::<OldTransactions>()
    |           -^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |           |
    |  _________creates a temporary value which is freed while still in use
    | |
184 | |             .get_batch(ids)
    | |___________________________- argument requires that borrow lasts for `'a`
...
187 |       }
    |       - temporary value is freed at the end of this statement
```

### Going forward
For this PR I suggest we don't go down this rabbit hole, but I'd be happy to do it as a follow-up if you think it's possible/interesting to explore it. Seems like a quite low prio though.